### DISCUSSION TITLE
feat: improve expo example

### DISCUSSION THREAD

**Discussion by juliusmarminge** [t3-oss/create-t3-turbo#967] [Ideas]
### Describe the feature you'd like to request

the idea of the expo example was never to have a perfect mobile setup from the start, but more to show the splitting possible.

but i tihnk we can add some stuff that will enhance the user experience of mobile devs checking this out

### Describe the solution you'd like to see

- [x] light/dark mode (#795)
- [ ] modals/drawers?
  - [ ] other navigation primitives that's commonly found in mobile apps?
- [ ] auth (#720)
  - maybe this can be refactored with api routes available directly in expo? that way we wouldn't need to deal with cross-origin stuff on the nextjs app? don't know if that makes sense though as you'd then have 2 server deployments ://
- [ ] keyboard avoiding when writing in input fields
- [ ] more??
- [ ] 
### Additional information

https://github.com/supabase-community/create-t3-turbo got some of these I believe

**Comment by EuanMorgan** [comment]
Could possibly add Apple sign in to this?
Just because its in Apple's ToS now that apps must offer it if they use any third party auth like the discord provider here

**Comment by juliusmarminge** [comment]
Yup - i just dont have an Apple developer account so cant test it 😆

**Comment by trevorpfiz** [comment]
I am working on this. thoughts on using [react-native-reusables](https://github.com/mrzachnugent/react-native-reusables)? best 'shadcn/ui for React Native' I have found.

**Reply by rrdesignweb** [reply]
Love this suggestion, I am looking at this too - gluestack is another option - https://github.com/gluestack/gluestack-ui ?

**Comment by trevorpfiz** [comment]
for the keyboard avoiding, [this thread](https://x.com/hirbod_dev/status/1668528361457451009?s=20) sums up the best options. it matches my experience.

we should probably just use [react-native-keyboard-aware-scroll-view](https://github.com/APSL/react-native-keyboard-aware-scroll-view) for a demo as it works with Expo Go? can recommend the other options in the FAQs?

**Comment by trevorpfiz** [comment]
> Could possibly add Apple sign in to this? Just because its in Apple's ToS now that apps must offer it if they use any third party auth like the discord provider here

I am trying to add Apple sign-in, but running into two issues:

1. https://github.com/nextauthjs/next-auth/issues/9989
2. https://github.com/nextauthjs/next-auth/discussions/6898#discussioncomment-8429480

**Comment by tconroy** [comment]
I think it'd be great if the `ui` package could be platform agnostic (work on web, ios, and android).

I think this would require switching to something like react-native-web, or maybe include react-native-reusables per @trevorpfiz's suggestion

**Reply by rrdesignweb** [reply]
Love this suggestion, I am looking at this too - gluestack is another option - https://github.com/gluestack/gluestack-ui ?
